src/sage/categories/rings.py

```diff
@@ -889,26 +922,23 @@ def ideal(self, *args, **kwds):
                     elif isinstance(first, (list, tuple, GeneratorType)):
                         gens = first
                     else:
-                        try:
-                            if self.has_coerce_map_from(first):
-                                gens = first.gens()  # we have a ring as argument
-                            elif isinstance(first, Element):
-                                gens = [first]
-                            else:
-                                raise ArithmeticError("there is no coercion from %s to %s" % (first, self))
-                        except TypeError:  # first may be a ring element
-                            pass
                         break
-            if coerce:
+
+            if not gens:
+                gens = [self.zero()]
+            elif coerce:
                 gens = [self(g) for g in gens]
+
             from sage.categories.principal_ideal_domains import PrincipalIdealDomains
             if self in PrincipalIdealDomains():
                 # Use GCD algorithm to obtain a principal ideal
                 g = gens[0]
                 if len(gens) == 1:
                     try:
-                        g = g.gcd(g)  # note: we set g = gcd(g, g) to "canonicalize" the generator: make polynomials monic, etc.
-                    except (AttributeError, NotImplementedError):
+                        # note: we set g = gcd(g, g) to "canonicalize" the generator:
+                        # make polynomials monic, etc.
+                        g = g.gcd(g)
+                    except (AttributeError, NotImplementedError, IndexError):
```

Comment:
It would be good to have a test for this line, in particular because `IndexError` was added.

Reply:
so far, I was not able to find a ring that pass through that case.

Reply:
https://en.wikipedia.org/wiki/Euclidean_domain has $\mathbb R[x,y]/(x^2+y^2+1)$, but sage does not recognise it as principal ideal domain (or does $\mathbb Q$ vs $\mathbb R$ make a difference?):
```
sage: R.<x,y> = PolynomialRing(QQ)
sage: Q = R.quotient(x^2+y^2+1)
sage: Q in PrincipalIdealDomains()
False
```
Also, if I made no mistake, $\mathbb Q(\sqrt{19})$ would constitute an example of a principal ideal domain which is not Euclidean, but sage does not know it (or do I misunderstand the code?):
```
sage: R.<x> = QQ[]
sage: K.<a> = QQ.extension(x^2 + 19); K
Number Field in a with defining polynomial x^2 + 19
sage: K
Number Field in a with defining polynomial x^2 + 19
sage: K in PrincipalIdealDomains()
True
sage: K in EuclideanDomains()
True
```

(Disclaimer: I know nothing about this kind of math)

So, I guess it's best to leave it in.